// @cloudinary/url-gen/actions/adjust.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var adjust = require('../adjust-509b6b72.cjs');
require('../FillLightAction-0fc49d64.cjs');
require('../Action-34aa7481.cjs');
require('../FlagQualifier-0e14a6c3.cjs');
require('../QualifierValue-e770d619.cjs');
require('../Qualifier-6633a22f.cjs');
require('../QualifierModel-0923d819.cjs');
require('../unsupportedError-74070138.cjs');
require('../RecolorAction-b6b06f88.cjs');
require('../OpacityAdjustAction-17f74b54.cjs');
require('../By3dLutAction-0e9ae162.cjs');
require('../ImproveAction-cdd16b5f.cjs');
require('../ReplaceColorAction-953def82.cjs');
require('../EffectActionWithLevel-3a4497d5.cjs');
require('../LeveledEffectAction-f3180bc9.cjs');
require('../SimpleEffectAction-96104f1f.cjs');
require('../internalConstants-6e675c29.cjs');
require('../EffectActionWithStrength-941b96c8.cjs');
require('../EffectActionWithBlend-7b564fbf.cjs');
require('../ViesusCorrectAdjustAction-99284761.cjs');



exports.Adjust = adjust.Adjust;
exports.autoBrightness = adjust.autoBrightness;
exports.autoColor = adjust.autoColor;
exports.autoContrast = adjust.autoContrast;
exports.blue = adjust.blue;
exports.brightness = adjust.brightness;
exports.brightnessHSB = adjust.brightnessHSB;
exports.by3dLut = adjust.by3dLut;
exports.contrast = adjust.contrast;
exports.fillLight = adjust.fillLight;
exports.gamma = adjust.gamma;
exports.green = adjust.green;
exports.hue = adjust.hue;
exports.improve = adjust.improve;
exports.opacity = adjust.opacity;
exports.opacityThreshold = adjust.opacityThreshold;
exports.recolor = adjust.recolor;
exports.red = adjust.red;
exports.replaceColor = adjust.replaceColor;
exports.saturation = adjust.saturation;
exports.sharpen = adjust.sharpen;
exports.tint = adjust.tint;
exports.unsharpMask = adjust.unsharpMask;
exports.vibrance = adjust.vibrance;
exports.viesusCorrect = adjust.viesusCorrect;
