// @cloudinary/transformation-builder-sdk/qualifiers/foregroundObject.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var tslib_es6 = require('../tslib.es6-5bb8bdb3.cjs');

/**
 * @memberOf Qualifiers
 * @description This namespace contains all the foreground objects used in the SDK
 * @namespace ForegroundObject
 * @example
 * import {ForegroundObject} from '@cloudinary/url-gen/qualifiers/foregroundObject'
 */
var FOREGROUND_OBJECT = {
    AIRPLANE: 'airplane',
    BUS: 'bus',
    DINING_TABLE: 'dining_table',
    SHEEP: 'sheep',
    BICYCLE: 'bicycle',
    CAR: 'car',
    DOG: 'dog',
    SOFA: 'sofa',
    BIRD: 'bird',
    CAT: 'cat',
    HORSE: 'horse',
    TRAIN: 'train',
    BOAT: 'boat',
    CHAIR: 'chair',
    PERSON: 'person',
    TV: 'tv',
    BOTTLE: 'bottle',
    COW: 'cow',
    POTTED_PLANT: 'potted_plant',
    MOTORBIKE: 'motorbike',
};
/**
 * @summary qualifier
 * @description Foreground object "airplane".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var airplane = function () { return FOREGROUND_OBJECT.AIRPLANE; };
/**
 * @summary qualifier
 * @description Foreground object "bus".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var bus = function () { return FOREGROUND_OBJECT.BUS; };
/**
 * @summary qualifier
 * @description Foreground object "dining_table".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var diningTable = function () { return FOREGROUND_OBJECT.DINING_TABLE; };
/**
 * @summary qualifier
 * @description Foreground object "sheep".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var sheep = function () { return FOREGROUND_OBJECT.SHEEP; };
/**
 * @summary qualifier
 * @description Foreground object "bicycle".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var bicycle = function () { return FOREGROUND_OBJECT.BICYCLE; };
/**
 * @summary qualifier
 * @description Foreground object "car".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var car = function () { return FOREGROUND_OBJECT.CAR; };
/**
 * @summary qualifier
 * @description Foreground object "dog".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var dog = function () { return FOREGROUND_OBJECT.DOG; };
/**
 * @summary qualifier
 * @description Foreground object "sofa".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var sofa = function () { return FOREGROUND_OBJECT.SOFA; };
/**
 * @summary qualifier
 * @description Foreground object "bird".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var bird = function () { return FOREGROUND_OBJECT.BIRD; };
/**
 * @summary qualifier
 * @description Foreground object "cat".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var cat = function () { return FOREGROUND_OBJECT.CAT; };
/**
 * @summary qualifier
 * @description Foreground object "horse".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var horse = function () { return FOREGROUND_OBJECT.HORSE; };
/**
 * @summary qualifier
 * @description Foreground object "train".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var train = function () { return FOREGROUND_OBJECT.TRAIN; };
/**
 * @summary qualifier
 * @description Foreground object "boat".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var boat = function () { return FOREGROUND_OBJECT.BOAT; };
/**
 * @summary qualifier
 * @description Foreground object "chair".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var chair = function () { return FOREGROUND_OBJECT.CHAIR; };
/**
 * @summary qualifier
 * @description Foreground object "person".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var person = function () { return FOREGROUND_OBJECT.PERSON; };
/**
 * @summary qualifier
 * @description Foreground object "tv".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var tv = function () { return FOREGROUND_OBJECT.TV; };
/**
 * @summary qualifier
 * @description Foreground object "bottle".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var bottle = function () { return FOREGROUND_OBJECT.BOTTLE; };
/**
 * @summary qualifier
 * @description Foreground object "cow".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var cow = function () { return FOREGROUND_OBJECT.COW; };
/**
 * @summary qualifier
 * @description Foreground object "potted_plant".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var pottedPlant = function () { return FOREGROUND_OBJECT.POTTED_PLANT; };
/**
 * @summary qualifier
 * @description Foreground object "motorbike".
 * @memberOf Qualifiers.ForegroundObject
 * @return {Qualifiers.ForegroundObject}
 */
var motorbike = function () { return FOREGROUND_OBJECT.MOTORBIKE; };
var ForegroundObject = tslib_es6.__assign(tslib_es6.__assign({}, FOREGROUND_OBJECT), { airplane: airplane,
    bus: bus,
    diningTable: diningTable,
    sheep: sheep,
    bicycle: bicycle,
    car: car,
    dog: dog,
    sofa: sofa,
    bird: bird,
    cat: cat,
    horse: horse,
    train: train,
    boat: boat,
    chair: chair,
    person: person,
    tv: tv,
    bottle: bottle,
    cow: cow,
    pottedPlant: pottedPlant,
    motorbike: motorbike });

exports.ForegroundObject = ForegroundObject;
exports.airplane = airplane;
exports.bicycle = bicycle;
exports.bird = bird;
exports.boat = boat;
exports.bottle = bottle;
exports.bus = bus;
exports.car = car;
exports.cat = cat;
exports.chair = chair;
exports.cow = cow;
exports.diningTable = diningTable;
exports.dog = dog;
exports.horse = horse;
exports.motorbike = motorbike;
exports.person = person;
exports.pottedPlant = pottedPlant;
exports.sheep = sheep;
exports.sofa = sofa;
exports.train = train;
exports.tv = tv;
